clients/spark/build.sbt

```diff
@@ -140,6 +140,7 @@ def getSharedLibraryDependencies(buildType: BuildType): Seq[ModuleID] = {
     "org.scalactic" %% "scalactic" % "3.2.9",
     "dev.failsafe" % "failsafe" % "3.2.4",
     "org.apache.hadoop" % "hadoop-distcp" % buildType.hadoopVersion,
+    "io.github.resilience4j" % "resilience4j-retry" % "1.7.0",
```

Comment:
Can we use [failsafe](https://failsafe.dev/), which is already included above?  I prefer to have a single retry mechanism if possible.